File: go/grind75/MaximumDepthofBinaryTree.go
package grind75

func maxDepth(root *TreeNode) int {
	if root == nil {
		return 0
	}
	depth := 1
	maxSub := max(maxDepth(root.Left), maxDepth(root.Right))
	return depth + maxSub
}

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

File: src/main/java/neetcode150/tree/MaximumDepthofBinaryTree.java
package neetcode150.tree;


public class MaximumDepthofBinaryTree {
    public int maxDepth(TreeNode root) {
        if (root == null) return 0;

        int dept = 1;
        int max = Math.max(maxDepth(root.left), maxDepth(root.right));
        return dept + max;
    }
}
